### PR TITLE
Visits without data

### LINKED ISSUE #73: Hide visits with no data checkbox
Request from Jack - hours and days are smooshing together for a client's version of the PoE and we would like to fix this for him ASAP. Projected deadline is mid-Jan

# Test notes
* A checkbox appears in the _Controls_ section that toggles the display of visits without data.

### PR DESCRIPTION
**Pull request by samussiah**
# Issues
Closes #73

### CODE REVIEW

File: src/charts/onPreprocess/setXoptions.js [hunk 1]
```diff
@@ -1,12 +1,20 @@
...
+        this.config.x,
+        this.config.time_cols.find(time_col => time_col.value_col === this.config.x.column)
+    );
+
+    //Remove visits without data from x-domain if x-type is ordinal.
+    if (this.config.x.type === 'ordinal') {
+        removeVisitsWithoutData.call(this);
+    }
+
+    //Delete domain setting if x-type is linear
```
Comment (jwildfire) [inline src/charts/onPreprocess/setXoptions.js:15]: Just curious: Why is this needed?

File: src/init.js [hunk 2]
```diff
@@ -22,7 +22,7 @@ export default function init(data) {
     layout.call(this);
 
     //Initialize charts.
-    multiply(this, this.data.raw, 'measure_unit');
+    multiply(this, this.data.raw, 'measure_unit', this.config.allMeasures);
```
Comment (jwildfire) [inline src/init.js]: So that cells will draw even when there is no data?

**Review by jwildfire — APPROVED** [review]
👍

File: src/charts/onPreprocess/setXoptions.js (hunk 1 above)
Comment (samussiah) [reply src/charts/onPreprocess/setXoptions.js:15]: Because the ordinal domain setting messes up the linear domain.

File: src/init.js (hunk 2 above)
Comment (samussiah) [reply src/init.js]: So the multiples draw in alphabetical order.  Otherwise they draw as they appear in the data.

**Comment by brittsikora** [comment]
Pass - this is working as expected in the settings-schema branch

**Review by brittsikora — APPROVED** [review]
